packages/coinstac-ui/app/render/components/user/user-edit-controller.jsx

```diff
@@ -0,0 +1,200 @@
+import { connect } from 'react-redux';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import crypto from 'crypto';
+import { notifySuccess, notifyError, writeLog } from '../../state/ducks/notifyAndLog';
+import { update } from '../../state/ducks/auth';
+import UserEdit from './user-edit';
+
+const CLOUDINARY_UPLOAD_PRESET = 'rxdgvbk9';
+const CLOUDINARY_API_KEY = '457127984681938';
+const CLOUDINARY_API_SECRET = 'mmx87Xdnl4FsSsRcU1Pi5hTEXy8';
+const CLOUDINARY_UPLOAD_URL = 'https://api.cloudinary.com/v1_1/cstacimg/upload';
+const CLOUDINARY_DELETE_URL = 'https://api.cloudinary.com/v1_1/cstacimg/image/destroy';
+
+
+class UserAccountController extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: null,
+      uploadedFileCloudinaryUrl: this.props.user.photo,
+      uploadedFileCloudinaryID: this.props.user.photoID,
+      uploadProgress: 0,
+      uploadProgressShow: false,
+    };
+  }
+
+  /**
+   * Handle new user edit submits.
+   *
+   * @todo  Improve form validation, move Redux action or middleware.
+   *
+   * @param  {object}    formData
+   * @param  {string}    formData.email
+   * @param  {string}    formData.institution
+   * @param  {string}    formData.id
+   * @param  {string}    formData.name
+   * @param  {string}    formData.photo
+   * @param  {string}    formData.photoID
+   * @param  {string}    formData.password
+   * @param  {string}    formData.username
+   * @return {undefined}
+   */
+  onSubmit = (formData) => {
+    let error;
+
+    if (!formData.id) {
+      error = 'ID required';
+    } else if (!formData.username) {
+      error = 'Username required';
+    } else if (!formData.email) {
+      error = 'Email required';
+    }
+
+    if (error) {
+      return this.handleUpdateError(error);
+    }
+
+    this.setState({ error: null })
+
+    return this.props.update(formData)
+      .then((res) => {
+        this.props.notifySuccess('User account changed');
+      });
+  }
+
+  // *********** Upload file to Cloudinary ******************** //
+  handleImageUpload = (file) => {
```

Comment:
this is all probably a lot easier w/ axios, but looks fine for now

Reply:
Yup, though formData makes it a bit cumbersome. I'll circle back to this in the future and refactor with axios.